polyply/src/apply_links.py

```diff
@@ -466,13 +321,23 @@ def run_molecule(self, meta_molecule):
         molecule = meta_molecule.molecule
         force_field = meta_molecule.force_field
 
-        for edge in meta_molecule.edges:
-            links, resids = _get_links(meta_molecule, edge)
-            for link, idxs in zip(links, resids):
-                try:
-                    apply_link_between_residues(molecule, link, idxs)
-                except MatchError:
-                    continue
+        for link in tqdm(force_field.links):
+            res_link = make_residue_graph(link, attrs=('order',))
```

Comment:
yes true

Reply:
actually this is a problem, because resname can be a choice object, which the residue graph maker cannot handle. Is that a bug? I think so.

In addition based on the order comparison matrix it is even not allowed to have residues with different order attributes. That is because an order attribute refers to a unique relative resid and two residues cannot have the same resid but different resnames. For example in the following link would never apply because '>' would always be say the resid needs to be larger in comparison to the first residue. But they are also interpreted to be the same  '>' if I interpret the order matrix correctly. It actually says in the doc-string even. 

```
[ link ]
[ angle ]
A >Q {resname: RES1} >P {resname: RES2}
```